### PR TITLE
fix: add alert report timeout limits

### PR DESCRIPTION
**Pull request by riahk**
### SUMMARY
- [x] Working Timeout and Grace Period fields should not be negative
- [x] Adds min value to modal input fields to prevent user from setting negative values

### TEST PLAN
Manual Test:
1. Open alert/report modal
2. Enter values for Working Timeout/Grace Period (where applicable)
3. Confirm that value cannot be set below 0

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by dpgaspar — COMMENTED** [review]
I think it makes sense to add a validation on the backend also:
https://github.com/apache/superset/blob/master/superset/reports/schemas.py#L162

```
fields.Integer(validate=[Range(min=1, error="Value must be greater than 0")])
```

**Comment by willbarrett** [comment]
Let's also add a test for this change.

**Review by amitmiran137 — APPROVED** [review]
Looks good.
Do we enforce this on the BE as well?

File: superset/reports/schemas.py [hunk 1]
```diff
@@ -160,7 +161,10 @@ class ReportSchedulePostSchema(Schema):
     validator_config_json = fields.Nested(ValidatorConfigJSONSchema)
     log_retention = fields.Integer(description=log_retention_description, example=90)
```
Comment (dpgaspar) [inline superset/reports/schemas.py:162]: Can we add the positive integer validation for `log_retention` and `working_timeout` on this PR?

File: superset/reports/schemas.py [hunk 2]
```diff
@@ -158,14 +159,22 @@ class ReportSchedulePostSchema(Schema):
...
+        description=log_retention_description,
+        example=90,
+        validate=[Range(min=1, error=_("Value must be greater than 0"))],
+    )
     grace_period = fields.Integer(
-        description=grace_period_description, example=60 * 60 * 4, default=60 * 60 * 4
+        description=grace_period_description,
+        example=60 * 60 * 4,
+        default=60 * 60 * 4,
+        validate=[Range(min=1, error=_("Value must be greater than 0"))],
```
Comment (willbarrett) [inline superset/reports/schemas.py:171]: I'm not sure if this is correct, as reports do not require a grace period.

**Review by willbarrett — COMMENTED** [review]
One comment on the python. Let's add some tests for the back-end validations as well.

File: superset/reports/schemas.py (hunk 2 above)
Comment (riahk) [reply superset/reports/schemas.py:171]: I may be mistaken, but I'm pretty sure we use this schema for both alerts and reports.

**Pull request merged**